image-engine/src/main/java/com/safehill/kcrypto/image_engine/model/ImageFilterArgs.kt

```diff
@@ -0,0 +1,15 @@
+package com.safehill.kcrypto.image_engine.model
+
+import android.net.Uri
+
+data class ImageFilterArgs(
+    val inputImage: Uri,
+    val type: Type
+) {
+
+    enum class Type(val id: String) {
+        Cuphead("cuphead"),
+        Mosaic("mosaic"),
+        StarryNight("starry-night"),
```

Comment:
What are the steps one should take to add another filter, beyond having the model ready?
Can you document that and minimise the changes? Can we consider merging the two enums: `ImageFilterArgs.Typ` and `MLModel`?

Reply:
I created a separate class to decouple filters from the actual ml models, for example you might want to add a filter that uses a combination of multiple ml models, or switch ml model based on device characteristics when using a given filter